joomla/joomla-cms
# [5.x] Template color scheme configuration

Pull Request for Issue #42134 .

### Summary of Changes

Add a Color scheme switch to Atum template. 
The PR allow to choose prefered color scheme based on OS setting, or fixed.
Instead of using media query (which is not possible to override) it uses data attribute (`data-bs-theme` for bootstrap).
Very similar to what @dgrammatiko tried in #42209 .

### New API
The following data attributes should be used within template:
`data-color-scheme-os` When template should follow OS setting
`data-color-scheme="light"` Use Light scheme
`data-color-scheme="dark"` Use Dark scheme

The combination `data-color-scheme-os data-color-scheme="dark"` means that the template will follow OS setting, but initialy will be used `dark` color scheme.

Addittionaly the PR introduce a new JavaScript event `joomla:color-scheme-change` which dispatched when color scheme changes.

All of this will allow to extensions easily follow to actual template state.
And in theory, should simplify Editor fixes for specified color scheme.

#### SCSS fixes needed

Following SCSS code does not work:
```scss
// Inherit parent with &
.foo{
  @if $enable-dark-mode {
    @include color-mode(dark) {
      &.bar {
        background-color: var(--template-bg-dark-90);
      }
    }
  }
}
// Child element
.foo{
  @if $enable-dark-mode {
    @include color-mode(dark) {
      .bar {
        background-color: var(--template-bg-dark-90);
      }
    }
  }
}
```
All such instance should use:
```scss
// For fisrt example
@if $enable-dark-mode {
  @include color-mode(dark) {
    .foo.bar {
      background-color: var(--template-bg-dark-90);
    }
  }
}
// For second example
@if $enable-dark-mode {
  @include color-mode(dark) {
    .foo .bar {
      background-color: var(--template-bg-dark-90);
    }
  }
}
```
(Or maybe someoen know better fixes?)

After the PR will be merged someone should look on it. Or can send PR against my PR.


### Testing Instructions
Apply patch run `npm install`.
Check the template Atum parameters `Dark mode`.
When its set to `Follow OS settings` then template color scheme should changes when OS color scheme changes.
When its set to `Use Light theme` or `Use Dark theme` then  template color scheme should stay Light or Dark, always.


### Link to documentations
Please select:
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/238
